MiniAn/minian_main.py

```diff
@@ -784,36 +784,28 @@ def save_minian_to_doric(
     res = Y - AC # residual images
 
     print(mn_defs.Messages.GEN_ROI_NAMES, flush = True)
-    names = []
-    usernames = []
-    for i in A.coords["unit_id"].values:
-        names.append("ROI"+str(i).zfill(4))
-        usernames.append("ROI {0}".format(i))
+    roi_ids           = A.coords["unit_id"].values
+    roi_dataset_names = [defs.DoricFile.Dataset.ROI.format(str(id_).zfill(4)) for id_ in roi_ids]
+    roi_usernames     = [defs.DoricFile.Dataset.ROI.format(id_) for id_ in roi_ids]
+    roi_names         = [defs.DoricFile.Dataset.ROI.format(id_) for id_ in roi_ids]
 
     with h5py.File(vname, 'a') as f:
 
         # Check if MiniAn results already exist
-        operationCount = ""
-        if vpath in f:
-            operations = [ name for name in f[vpath] if mn_defs.DoricFile.Group.ROISIGNALS in name ]
-            if len(operations) > 0:
-                operationCount = str(len(operations))
-                for operation in operations:
-                    operationAttrs = utils.load_attributes(f, f"{vpath}/{operation}")
-                    if utils.merge_params(params_doric, params_source) == operationAttrs:
-                        if(len(operation) == len(mn_defs.DoricFile.Group.ROISIGNALS)):
-                            operationCount = ""
-                        else:
-                            operationCount = operation[-1]
-
-                        break
+        operationCount = utils.operation_count(vpath, f, mn_defs.DoricFile.Group.ROISIGNALS, params_doric, params_source)
```

Comment:
I also created a function to generate the operation count. I don't know why I didn't do that before